nntrainer/models/neuralnet.cpp

```diff
@@ -245,13 +201,14 @@ sharedConstTensors NeuralNetwork::forwarding(bool training) {
 sharedConstTensors NeuralNetwork::forwarding(sharedConstTensors input,
                                              sharedConstTensors label,
                                              bool training) {
-
-  NNTR_THROW_IF(input[0]->batch() != batch_size ||
-                  (!label.empty() && label[0]->batch() != batch_size),
+  auto current_batch = model_graph.getBatchSize();
```

Comment:
this creates two cache values for batch size - one in props::BatchSize and another in model_graph.batch_size.
I don't think network_graph needs to cache batch_size.

Reply:
so, props::BatchSize is a hint (for training). Single source of true value lies in model_graph.getBatchSize(). With this setup, model::inference doesn't change training batch size.

This will be needed if we are going to change only validation batchsize as well.

Reply:
if the batch size in the graph created (model graph) is cached for training, then it should be `training_batch_size`.
Given model_graph is abstracted from training/validation/inference, storing a batch_size which mismatches with the current set batch_size in the model_graph is not a good thing.
If you intend to cache the various types of batch size, IMO they are best done in neuralnetwork class.

Reply:
graph always stores a batch size with currently set batch size.
We can have various types of batch size cached in neuralnet (in the future) but that doesn't mean it removes the need of batchsize in the graph.
We need to query the batchsize from the model graph.

Reply:
the graph also needs to store the current execution mode in which it is (train, inference or validation). In the current implementation, graph is kept abstract from the execution mode, but to store the batch size, graph must then store this as well.

Also, if a graph/model runs inference with batch size 2, then the batch size of the graph/model is 2 (although the batch size of the last train might be 32).

Reply:
we discussed offline to change name to training batchsize to make it clear.